### PR TITLE
[WIP] Support for craft

### PR DESCRIPTION
**Pull request by james2432**
Related to Issue #1697
Craft icons added
SQL update for craft to be selected

Done:

- [x] tailor
- [x] plumber
- [x] shoemaker
- [x] winery
- [x] photographer

What's left (After feed back):
Colour of icons:
- [ ] Brown
- [ ] Purple
- [ ] Orange
-----------------------------
Icons:
- [ ] metal_construction (icon needed)
- [ ] gardener (consensus on adding it?)
- [ ] confectionery (consensus on adding it?)
- [ ] brewery (no stroke to be used)
- [ ] carpenter (no stroke to be used)
- [ ] electrician (no stroke to be used)
 
From https://www.openstreetmap.org/#map=19/44.25915/-76.58217 (node)
image

image

From https://www.openstreetmap.org/#map=19/42.50290/-83.18880 (way):
image

image

image

### CODE REVIEW

**Comment by Tomasz-W** [comment]
I'm not sure about amenity-brown color for this feature, maybe shop-violet would fit better? What do you think?

**Comment by james2432** [comment]
Was following what was said here: https://github.com/gravitystorm/openstreetmap-carto/issues/1697#issuecomment-225014363

I can change it, it's pretty trivial compared to the rest

**Comment by james2432** [comment]
Purple would look like this:
image

**Comment by kocio-pl** [comment]
We should discuss the tagging question of craft/shop=tailor on Tagging list before merging - it's not clear for me which one should we use (or maybe both).

**Comment by polarbearing** [comment]
craft/shop=tailor has a semantic difference that would justify to support both. It also has a tagging issue that shop was used for quite long before craft=* generally came up.

**Comment by kocio-pl** [comment]
As a tip for discussion - here is the dynamic of uses of both tags:
taghistory 15

**Comment by jragusa** [comment]
`craft=dressmaker` should be also included. According to the [wiki](https://wiki.openstreetmap.org/wiki/Tag:craft=dressmaker), tailor is for men and dressmaker for women

**Comment by james2432** [comment]
@jragusa Yet the wiki page has: "It has been proposed that this page or section be merged with Tag:craft=tailor"

**Comment by Tomasz-W** [comment]
@james2432 Can you upload test renderings of rest of icons?

**Comment by james2432** [comment]
have to rework the query as this would require database re-import. I need to use hstore via the tags->'craft' instead

**Comment by matthijsmelissen** [comment]
@james2432 Are you planning to continue working on this? This would something that's definitely nice to have. Is there any input you need from us?

**Comment by james2432** [comment]
yeah was planning on working on this tonight, was busy with life stuff

**Comment by james2432** [comment]
I do need icons for metal_construction and gardener and whether or not confectionary has a consensus on being added or not

**Comment by james2432** [comment]
Brown:
image

grey:
image

Orange(gastronomy)#c64803:
image

**Comment by meased** [comment]
What do you think about using the new office blue color?

**Comment by james2432** [comment]
SVGs are not myself that created them nor do I know how to edit SVGs without messing them up.

I'll look into the other stuff later

File: project.mml [hunk 1]
```diff
@@ -1543,7 +1546,8 @@ Layer:
             OR tags->'memorial' IN ('plaque')
             OR military IN ('bunker')
             OR highway IN ('bus_stop', 'elevator', 'traffic_signals')
-            OR (power = 'generator' AND (tags @> '"generator:source"=>wind' OR tags @> 'power_source=>wind'))
+            OR (power = 'generator' AND (tags @> '"generator:source"=>wind' OR tags @> 'power_source=>wind')
+            OR tags->'craft' IN ('brewery','carpenter','electrician','plumber','shoemaker','winery','photographer','tailor'))
```
[the remark this answers is not in the document]
Comment (james2432) [reply project.mml]: https://taginfo.openstreetmap.org/keys/craft#values no they don't very very very small subset.

File: project.mml [hunk 2]
```diff
@@ -1609,7 +1613,8 @@ Layer:
               'tourism_' || CASE WHEN tourism IN ('viewpoint') THEN tourism ELSE NULL END,
               'barrier_' || CASE WHEN barrier IN ('toll_booth') THEN barrier ELSE NULL END,
               'man_made_' || CASE WHEN man_made IN ('cross') THEN man_made ELSE NULL END,
-              'historic_' || CASE WHEN historic IN ('wayside_cross', 'wayside_shrine') THEN historic ELSE NULL END
+              'historic_' || CASE WHEN historic IN ('wayside_cross', 'wayside_shrine') THEN historic ELSE NULL END,
+              'craft' || CASE WHEN tags->'craft' IN ('brewery','carpenter','electrician','plumber','shoemaker','winery','photographer','tailor') THEN '' ELSE NULL END
```
[the remark this answers is not in the document]
Comment (james2432) [reply project.mml]: Same comments as above

File: project.mml [hunk 3]
```diff
@@ -1644,6 +1649,8 @@ Layer:
             tags->'tower:construction' as "tower:construction",
             tags->'tower:type' as "tower:type",
             tags->'castle_type' as castle_type,
+            CASE WHEN tags->'craft' IN ('brewery','carpenter','electrician','plumber','shoemaker','winery','photographer','tailor') THEN tags->'craft'
+                               ELSE '' END AS craft,
```
[the remark this answers is not in the document]
Comment (james2432) [reply project.mml]: Same comments as above

File: project.mml [hunk 4]
```diff
@@ -1676,7 +1683,8 @@ Layer:
             OR tags @> 'emergency=>phone'
             OR highway IN ('bus_stop', 'elevator', 'traffic_signals')
             OR tags @> 'ford=>yes' OR tags @> 'ford=>stepping_stones'
-            OR (power = 'generator' AND (tags @> '"generator:source"=>wind' OR tags @> 'power_source=>wind'))
+            OR (power = 'generator' AND (tags @> '"generator:source"=>wind' OR tags @> 'power_source=>wind')
+            OR tags->'craft' IN ('brewery','carpenter','electrician','plumber','shoemaker','winery','photographer','tailor'))
```
[the remark this answers is not in the document]
Comment (james2432) [reply project.mml]: Same comments as above

File: project.mml [hunk 5]
```diff
@@ -2124,7 +2134,8 @@ Layer:
               OR highway IN ('services', 'rest_area', 'bus_stop', 'elevator')
               OR power IN ('plant', 'station', 'generator', 'sub_station', 'substation')
               OR boundary IN ('national_park')
-              OR waterway IN ('dam', 'dock'))
+              OR waterway IN ('dam', 'dock')
+              OR tags->'craft' in ('brewery','carpenter','electrician','plumber','shoemaker','winery','photographer','tailor'))
```
[the remark this answers is not in the document]
Comment (james2432) [reply project.mml]: Same comments as above

File: symbols/craft/brewery.svg [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+     style="opacity:1;fill:#000000;fill-opacity:1;fill-rule:evenodd;stroke:none;stroke-width:0.9391678;stroke-linecap:square;stroke-linejoin:round;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1;paint-order:stroke fill markers" />
```
[the remark this answers is not in the document]
Comment (james2432) [reply symbols/craft/brewery.svg]: @Tomasz-W

File: symbols/craft/carpenter.svg [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+     style="opacity:1;fill:#000000;fill-opacity:1;fill-rule:evenodd;stroke:none;stroke-width:0.90670317;stroke-linecap:square;stroke-linejoin:round;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1;paint-order:stroke fill markers" />
```
[the remark this answers is not in the document]
Comment (james2432) [reply symbols/craft/carpenter.svg]: @Tomasz-W

File: symbols/craft/carpenter.svg [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  <path
+     id="rect231-1"
+     d="M 6.6484375 0.00390625 C 5.5413047 0.042041095 4.3646915 0.29389461 2.8671875 1.4140625 L 2.8652344 1.4121094 L 2.8554688 1.421875 C 2.8534608 1.4233816 2.8516185 1.4242716 2.8496094 1.4257812 L 2.8515625 1.4277344 L 1.4414062 2.859375 L 1.75 3.1679688 L 1.2675781 3.6523438 L 0.94921875 3.3320312 L 0 4.2851562 L 1.4355469 5.7265625 L 2.3847656 4.7734375 L 2.078125 4.4648438 L 2.5625 3.9804688 L 12.576172 14 L 14 12.552734 L 4.2890625 2.8359375 C 4.251405 1.4117303 6.6062529 0.40039063 7.7421875 0.40039062 C 7.7444975 0.13678931 7.7460938 0.24593006 7.7460938 0.01171875 C 7.3814942 0.002018925 7.0189646 -0.0088564409 6.6484375 0.00390625 z "
+     style="opacity:1;fill:#000000;fill-opacity:1;fill-rule:evenodd;stroke:none;stroke-width:0.47358352;stroke-linecap:square;stroke-linejoin:round;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1;paint-order:stroke fill markers" />
```
[the remark this answers is not in the document]
Comment (james2432) [reply symbols/craft/carpenter.svg]: @Tomasz-W

File: symbols/craft/electrician.svg [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+     style="opacity:1;fill:#000000;fill-opacity:1;fill-rule:evenodd;stroke:none;stroke-width:1.18225479;stroke-linecap:square;stroke-linejoin:round;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1;paint-order:stroke fill markers" />
```
[the remark this answers is not in the document]
Comment (james2432) [reply symbols/craft/electrician.svg]: @Tomasz-W

**Comment by pnorman** [comment]
Our SVG requirements are [in CONTRIBUTING.md](https://github.com/gravitystorm/openstreetmap-carto/blob/master/CONTRIBUTING.md#map-icon-guidelines)

**Comment by Tomasz-W** [comment]
Just for the note: these strokes are only settings, which are placed by Inkscape automatically

**Comment by hikemaniac** [comment]
I just pushed a PR to @james2432 repo which optimizes the SVGs and removes all the strokes. See james2432/openstreetmap-carto#1

**Comment by Tomasz-W** [comment]
@hikemaniac Meanwhile, I fixed icons to pixel aligned versions. See https://github.com/gravitystorm/openstreetmap-carto/issues/1697#issuecomment-416043315. Actually I should post this commment in this PR, sorry for inconvenience.

**Comment by hikemaniac** [comment]
@Tomasz-W This is great! 👍 Sorry, I didn't see the updated icons. Should I update the icons on this PR?

**Comment by james2432** [comment]
might as well,  i'll just merge it if theres a PR on my PR(lol)

**Comment by hikemaniac** [comment]
I submitted another pull request, which updates the icons to the new pixel aligned versions. See james2432/openstreetmap-carto#2

**Comment by Tomasz-W** [comment]
@hikemaniac Thanks :) Now we need a test renderings of these icons in _amenity-brown_. Are you interested in making them?

**Comment by polarbearing** [comment]
Was the colour decided already? amenity-brown would be my least desirable here. 
shop-purple makes more sense, e.g. hairdressers have this colour. 
office-blue would work as well, in particular as many of them are just the headquarters. You rarely bring your clogged kitchen sink to the plumber, do you?

**Comment by Tomasz-W** [comment]
I think shop-violet might be confusing as we go to "craft" places not to buy some good, but to buy some service (for repairing or constructing something), that's why I proposed amenity-brown. Earlier in the discussion I considered also office-blue for the same reason which you mentioned above, so I agree that it could be tested also.

**Comment by meased** [comment]
I'd like to see a test of office blue because it fits the category of "a place of business without a public street front".

**Comment by polarbearing** [comment]
Consequently you would need to move services like hair dresser, nail studio (currently violet) and car repair (currently brown) to blue. Which might be a good idea.

**Comment by kocio-pl** [comment]
But hair dresser, nail studio and car repair are not "a place of business without a public street front". I still think they should be all brown.

**Comment by polarbearing** [comment]
> not ... without

Double negations make the statement hard to read ;-)
Well, probably you can find some property for any pair of two objects that groups them into the same category. How is a public street front defined then?

**Comment by dieterdreist** [comment]
2018-09-11 6:49 GMT+02:00 kocio-pl <notifications@github.com>:

> But hair dresser, nail studio and car repair are not "a place of business
> without a public street front". I still think they should be all brown.
>

I would stick to the osm classification here, "shop" (selling goods or
services) in one color. Amenities in (at least one) different color(s).

**Comment by dieterdreist** [comment]
craft on a shop is like a qualifier (shop=bakery & craft=bakery) means a
place that produces and sells bread, other crafts may be like shops but
without a shop tag.

craft on it's own when it is not shop-like is not very interesting for a
general purpose map, but it is motivating for the mappers anyway. Brown
does not hit it for these cases (amenity color), I would go for the office
color or a variation of it, and "thin" labels / late rendering (according
to size, big craft places are interesting for slightly earlier labeling
like anything big that dominates the ground situation)

**Comment by Adamant36** [comment]
@tomasz-W, you want to give a quick, updated summary on what still needs to be done so I don't have to read through 4 months worth of messages since I wasn't involved in the original discussion? Once I know exactly whats left, I'll get to work on it.

**Comment by Tomasz-W** [comment]
Sure. Please test these icons in _amenity-brown_/ _office-blue_ versions:

* https://gist.github.com/Tomasz-W/4c4febd2f19dbfd8ec201100b54946ea (tailor icon can be used in the same colour also for `shop=tailor`)
* https://github.com/gravitystorm/openstreetmap-carto/blob/master/symbols/shop/photo.svg for `craft=photographer`
* https://github.com/gravitystorm/openstreetmap-carto/blob/master/symbols/shop/confectionery.svg for `craft=confectionery`
* https://github.com/gravitystorm/openstreetmap-carto/blob/master/symbols/shop/garden_centre.svg for `craft=gardener`

**Comment by matkoniecz** [comment]
@james2432 Are you planning to update this PR?

**Comment by james2432** [comment]
yes, I just have to put myself back where I was

**Comment by Adamant36** [comment]
James2432, how come you closed it?

**Pull request closed without merging**